### ISSUE TITLE
Vmalert: use "-datasource.url=127.0.0.1:8429" config panic

### ISSUE THREAD

**Issue by yincongcyincong** [VictoriaMetrics/VictoriaMetrics#6740]
### Describe the bug

when i start vmalert use blow config
```
vmalert -datasource.url=127.0.0.1:8429 -rule=/Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/test.yaml -notifier.url=http://localhost:9093
```
at first , it's run normally. but a few seconds ago.  i get a panic
```
panic: BUG: unexpected error from http.NewRequest("127.0.0.1:8429"): parse "127.0.0.1:8429": first path segment in URL cannot contain colon

goroutine 38 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0x102cbd0e1, 0x5}, {0x1400012c5a0, 0x84}, 0x4)
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/lib/logger/logger.go:301 +0x1420
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0x102cbd0e1, 0x5}, {0x102cddf24, 0x32}, {0x14000036768, 0x2, 0x2})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/lib/logger/logger.go:141 +0xb4
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel({0x102cbd0e1, 0x5}, {0x102cddf24, 0x32}, {0x14000036768, 0x2, 0x2})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/lib/logger/logger.go:133 +0x5c
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf({0x102cddf24, 0x32}, {0x14000036768, 0x2, 0x2})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/lib/logger/logger.go:129 +0x54
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/datasource.(*VMStorage).newRequest(0x1400015a070, {0x102f07ea8, 0x14000280190})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/datasource/vm.go:259 +0x1cc
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/datasource.(*VMStorage).newQueryRequest(0x1400015a070, {0x102f07ea8, 0x14000280190}, {0x1400020a180, 0x2}, {0x0, 0xede3d1584, 0x103288f00})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/datasource/vm.go:241 +0x60
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/datasource.(*VMStorage).Query(0x1400015a070, {0x102f07ea8, 0x14000280190}, {0x1400020a180, 0x2}, {0x0, 0xede3d1584, 0x103288f00})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/datasource/vm.go:138 +0x7c
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/rule.(*AlertingRule).exec(0x14000148000, {0x102f07ea8, 0x14000280190}, {0x0, 0xede3d1584, 0x103288f00}, 0x0)
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/rule/alerting.go:368 +0xcc
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/rule.(*executor).exec(0x140003f67e0, {0x102f07ea8, 0x14000280190}, {0x102f08340, 0x14000148000}, {0x0, 0xede3d1584, 0x103288f00}, 0x37e11d6000, 0x0)
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/rule/group.go:672 +0xa0
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/rule.(*executor).execConcurrently(0x140003f67e0, {0x102f07ea8, 0x14000280190}, {0x1400029e190, 0x1, 0x1}, {0x0, 0xede3d1584, 0x103288f00}, 0x1, ...)
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/rule/group.go:638 +0x2dc
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/rule.(*Group).Start.func2({0x102f07ea8, 0x14000280190}, {0x0, 0xede3d1584, 0x103288f00})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/rule/group.go:361 +0x1d8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/rule.(*Group).Start(0x14000288100, {0x102f07ea8, 0x140002100f0}, 0x140001700a0, {0x0, 0x0}, {0x0, 0x0})
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/rule/group.go:377 +0x570
main.(*manager).startGroup.func1()
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/manager.go:90 +0x128
created by main.(*manager).startGroup in goroutine 1
        /Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/app/vmalert/manager.go:87 +0x138
Exiting.

```
i think the expected behavior is vmalert can't start up.

### To Reproduce

```
vmalert -datasource.url=127.0.0.1:8429 -rule=/Users/yincong/go/src/github.com/yincongcyincong/VictoriaMetrics/test.yaml -notifier.url=http://localhost:9093
```

### Version

lastest master

**Comment by jiekun** [comment]
Maybe you should try `-datasource.url=http://127.0.0.1:8429`?
https://docs.victoriametrics.com/vmalert/

**Comment by f41gh7** [comment]
panic caused by incorrect value at flag`-datasource.url=127.0.0.1:8429` flag. It must have scheme - `http` -> `http://-datasource.url=127.0.0.1:8429`. 

 Currently, there is a validation problem for any flag that supposed to be `URL`, that used in any vm components.

 I'd propose to introduce new flag  - `URL`. That performs url validation and provides clear error to the user.

**Comment by f41gh7** [comment]
@Haleygo @hagen1778 WDYT?

**Comment by hagen1778** [comment]
@yincongcyincong the issue was addressed in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6820 and will be included into the next release.

**Comment by hagen1778** [comment]
This bugfix was included into [v1.103.0](https://docs.victoriametrics.com/changelog/#v11030), [v1.102.2](https://docs.victoriametrics.com/changelog/#v11022) and [1.97.7](https://docs.victoriametrics.com/changelog/#v1977) releases.
